packages/app/src/composables/useWallet.ts

```diff
@@ -31,6 +31,7 @@ type UseWallet = ToRefs<WalletState> & {
 
   getL1Signer: () => Promise<L1Signer>;
   getL2Signer: () => Promise<Signer>;
+  getEthereumProvider: () => Promise<BaseProvider | undefined>;
```

Comment:
Even when nullables are undefinables are kind of the same, I believe that using `BaseProvider | null` it's a little bit better. The main reason is that null denotates intention, while undefined can totally be a mistake like a variable not initialized or accessing an empty property of an object.